geo/src/algorithm/rotate.rs

```diff
@@ -194,7 +194,7 @@ where
 {
     /// Rotate the contained Polygons about their centroids by the given number of degrees
     fn rotate(&self, angle: T) -> Self {
-        MultiPolygon(self.iter().map(|poly| poly.rotate(angle)).collect())
+        MultiPolygon(self.iter().map(|poly| poly.rotate_around_point(angle, self.centroid().unwrap())).collect())
```

Comment:
Maybe?

```suggestion
        let centroid = self.centroid().unwrap();
        MultiPolygon(self.iter().map(|poly| poly.rotate_around_point(&centroid)).collect())
```

The `unwrap()` doesn't seem ideal, maybe it should return a clone of `self` there's no centroid?

Reply:
hmm, thinking about this more it seems like if we can't find a centroid, then we should probably return a None, which would require changing the function signature to return an option (as we find with `centroid()`)

Reply:
Or put another way, is there a satisfactory failure state if a centroid can't be returned?

Reply:
note `rotate` for polygon does do an `unwrap` on the centroid, which is why I did the same.  is that not a good precedent to follow?

```rust
fn rotate(&self, angle: T) -> Self {
        // if a polygon has holes, use the centroid of its outer shell as the rotation origin
        let centroid = if self.interiors().is_empty() {
            self.centroid().unwrap()
        } else {
            self.exterior().centroid().unwrap()
        };
        Polygon::new(
            rotate_many(angle, centroid, self.exterior().points_iter()).collect(),
            self.interiors()
                .iter()
                .map(|ring| rotate_many(angle, centroid, ring.points_iter()).collect())
                .collect(),
        )
    }
```

Reply:
I can't check right now and I'm not familiar with the code (there are probably some traits involved?) but it seems like you should be able to rotate e.g. an empty geometry.

Reply:
(I have changed along the lines you suggested, but I confess I'm not familiar enough with a rust api user's expectations to know what we should do.  My first instinct is to expect parity between `Multipolygon` and `Polygon`, (i.e. both should return `Option<Point>` or both should return `Point`)

Reply:
I agree, being consistent is more important here.

Reply:
Yeah I think your thought about the empty geometry is probably right. Looking at JTS centroid will return null in some cases (presumably for an empty geometry). Since the trait is shared across all types centroid always returns an option.

An empty multipolygon is a good example where a centroid perhaps wouldn't succeed and returning a clone of self would be suitable. I'll add a a test or two to test those edge cases (idk off hand if you can make an empty multipolygon but I'll try)